File: diarista/src/main/java/com/sistema/diarista/model/entity/Agendamento.java
package com.sistema.diarista.model.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.lang.Nullable;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class Agendamento extends GenericEntity<Long>{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date criacao;
    private Date dataRealizada;
    private Integer estrelas;
    private String comentario;
    private Date dataAgendada;
    @ManyToOne
    @JoinColumn(name = "fk_diarista_id")
    private Diarista diarista;
    @ManyToOne
    @JoinColumn(name = "fk_cliente_id")
    private Cliente cliente;
}

File: Diarista-View/src/app/interfaces/agendamento.ts
import { ICliente } from "./cliente";
import { Diarista } from "./diarista";

export interface Agendamento {
  id: number,
  criacao: string,
  dataRealizada: string,
  dataAgendada: string,
  dataAgendadaStr: string,
  cliente: ICliente,
  diarista: Diarista,
  estrelas: number,
  comentario: string,
}
